### PR TITLE
[DBEX] log when Form 0781 document upload fails

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

**Objective:** Implements a system for periodically polling the Lighthouse [uploads/status](https://dev-developer.va.gov/explore/api/benefits-documents/docs?version=current) endpoint to confirm the status of supplemental documents submitted to Lighthouse as part of a Form 526 Submission. The purpose of this system is to surface successes and failures that happen in the Form 526 process after being forwarded to Lighthouse for processing.

- Add logging for form 0781 type documents that are reported by Lighthouse to have failed upload (includes test)
- Add `form0781_types?` instance method to `Lighthouse526DocumentUpload` (includes test)
- Refactor `update_document_status` helper method in `UpdateDocumentsStatusService`
- *This work is behind a feature toggle (flipper): ~YES~/**NO***
- Responsible team: Disability Benefits Experience Team 2 (dBeX Carbs 🥖)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-api/pull/17492
- https://github.com/department-of-veterans-affairs/vets-api/pull/17473
- https://github.com/department-of-veterans-affairs/vets-api/pull/17477
- https://github.com/department-of-veterans-affairs/vets-api/pull/17439
- https://github.com/department-of-veterans-affairs/vets-api/pull/17405
- https://github.com/department-of-veterans-affairs/vets-api/pull/17407
- https://github.com/department-of-veterans-affairs/vets-api/pull/15964
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87626
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87621
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87619
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/75804

## Testing done

- [X] *New code is covered by unit tests*

## What areas of the site does it impact?

- non-user facing, asynchronous job tracking the status of uploaded documents submitted in conjunction with Form 526EZ 

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X]  Feature/bug has a monitor built into Datadog (if applicable)
  - https://vagov.ddog-gov.com/monitors/244896?view=spans
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by tblackwe — APPROVED** [review]

File: lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb [hunk 1]
```diff
@@ -64,30 +64,52 @@ def update_documents_status
...
+      def status_updater(status, document_upload)
+        # UploadStatusUpdater encapsulates all parsing of a status response from Lighthouse
+        @status_updater ||= BenefitsDocuments::Form526::UploadStatusUpdater.new(status, document_upload)
+      end
+
+      def statsd_document_base_key(statsd_document_type_key)
+        @statsd_document_base_key ||= "#{STATSD_BASE_KEY}.#{statsd_document_type_key}"
+      end
+
+      def log_failure(document_upload)
```
Comment (rmtolmach) [inline lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb:93]: Nice. I like that you pulled out some logic from`update_document_status` and made these methods 🎉

File: lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb [hunk 2]
```diff
@@ -64,30 +64,52 @@ def update_documents_status
...
+        # Because Lighthouse's processing steps are subject to change, these metrics must be dynamic.
+        # Currently, this should return either CLAIMS_EVIDENCE or BENEFITS_GATEWAY_SERVICE
+        failure_step = @status_updater.get_failure_step
+
+        # ex. 'api.form526.lighthouse_document_upload_processing_status.bdd_instructions.failed.claims_evidence'
+        StatsD.increment("#{@statsd_document_base_key}.#{STATSD_DOCUMENT_FAILED_KEY}.#{failure_step.downcase}")
+        return unless document_upload.form0781_types?
+
+        submission = document_upload.form526_submission
+        Rails.logger.info(
```
Comment (rmtolmach) [inline lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb:103]: Even though it's a failure, you want this at the `info` level?

**Review by rmtolmach — APPROVED** [review]
LGTM as long as the `info` logger is the logging level you want. Otherwise, `@` me and I'll re-review if you make any changes.

File: lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb (hunk 2 above)
Comment (freeheeling) [reply lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb]: @rmtolmach Considering your posed question, in [ef9e562](https://github.com/department-of-veterans-affairs/vets-api/pull/17625/commits/ef9e5628a44db1c35eb09b609dc9d2eb41d3c373), I've updated the log level to `warn`. Thanks!

**Comment by freeheeling** [comment]
@rmtolmach Reposting...

> Considering your posed question, in [ef9e562](https://github.com/department-of-veterans-affairs/vets-api/pull/17625/commits/ef9e5628a44db1c35eb09b609dc9d2eb41d3c373), I've updated the log level to warn. Thanks!

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**